src/routes/settings/components/Settings.js

```diff
@@ -0,0 +1,197 @@
+    height: ${StyleSheet.hairlineWidth};
+`
+
+const SectionHeaderContainer = styled.View`
+`
+
+const SectionHeader = styled.Text`
+    padding: 20px;
+    color: ${props => props.theme.colors.textColor};
+`
+
+export default class Settings extends React.Component<PropsType, StateType> {
+  sections = () => {
+    const {t, language} = this.props
+
+    return ([
+      {
+        title: 'Placeholder',
+        data: [
+          {
+            title: 'Placeholder',
+            description: 'Placeholder'
+          },
+          {
+            title: 'Placeholder',
+            description: 'Placeholder'
+          }
+        ]
+      },
+      {
+        data: [
+          {
+            title: t('troubleshooting'),
+            description: t('troubleshootingDescription'),
+            hasSwitch: true,
+            onPress: () => { this.toggleErrorTracking() }
+          },
+          {
+            title: t('about'),
+            onPress: () => {
+              if (language === 'de') {
+                Linking.openURL('https://integreat-app.de/')
+              } else {
+                Linking.openURL('https://integreat-app.de/en/')
+              }
+            }
+          },
+          {
+            title: t('privacyPolicy'),
+            onPress: () => {
+              if (language === 'de') {
+                Linking.openURL('https://integreat-app.de/datenschutz-webseite/')
+              } else {
+                Linking.openURL('https://integreat-app.de/en/privacy-website/')
+              }
+            }
+          },
+          {
+            title: t('version', {version: '??'})
+          },
+          {
+            title: t('openSourceLicenses'),
+            onPress: () => { console.warn('Not yet implemented.') }
+          }
+        ]
+      }
+    ])
+  }
+
+  constructor (props: PropsType) {
+    super(props)
+
+    this.state = {settingsLoaded: false, settings: {errorTracking: false, test: false}}
+
+    this.loadSettings()
+  }
+
+  async loadSettings () {
+    try {
+      const settingsKeys = Object.keys(this.state.settings)
+      const settingsArray = await AsyncStorage.multiGet(settingsKeys)
+
+      const settings = reduce(settingsArray,
+        (accumulator, [key, value]) => {
+          accumulator[key] = JSON.parse(value)
+          return accumulator
+        },
+        {})
+
+      this.setState(state => ({...state, settingsLoaded: true, settings}))
+    } catch (e) {
+      console.error('Failed to load settings.')
+    }
+  }
+
+  async setSetting (changeSetting: SettingsType => $Shape<SettingsType>) {
+    this.setState(state => {
+      const newSettings = changeSetting(state.settings)
+      return ({...state, settings: {...state.settings, ...newSettings}})
+    })
+
+    const settingsArray = reduce(changeSetting(this.state.settings),
+      (accumulator, value, key) => {
+        accumulator.push([key, JSON.stringify(value)])
+        return accumulator
+      },
+      [])
```

Comment:
With lodash you can do 
```js
const settingsArray = toPairs(mapValues(changeSetting(this.state.settings), value => JSON.stringify(value)))
```
Also you should probably do the AsyncStorage stuff in the callback of `setState`, right? Or is it not possible with promises?
What do you think of https://medium.com/front-end-weekly/async-await-with-react-lifecycle-methods-802e7760d802

Reply:
_Note if you do return a promise from the lifecycle methods it will be not be awaited._

So you can return it but it will not make a difference